### PR TITLE
WIP: implement redis backup fetch

### PR DESCRIPTION
**Pull request by prolomova**

### CODE REVIEW

**Comment by proggga** [comment]
Hi, we are going to improve the redis part of WAL-G, so we can merge this PR right now because we want to rewrite this code. If you want, you can close this PR because we will make these features in the near future.

File: docker/redis_tests/scripts/tests/redis_full_backup_test.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+actual_output=$(redis-cli get key)
+
+if [ "$actual_output" != "$expected_output" ]; then
+  echo "Error: actual output doesn't match expected output"
+  echo "Expected output: $expected_output"
+  echo "Actual output: $actual_output"
+  exit 1
+fi
+
+redis-cli shutdown
```
Comment (proggga) [inline docker/redis_tests/scripts/tests/redis_full_backup_test.sh:40]: nitpick: pls add newline at the end

File: internal/wal_fetch_handler.go [hunk 2]
```diff
@@ -128,6 +128,29 @@ func checkWALFileMagic(prefetched string) error {
...
+	for _, decompressor := range compression.Decompressors {
+		archiveReader, exists, err := TryDownloadFile(backup.BaseBackupFolder, backup.Name)
+		if err != nil {
+			return err
+		}
+		if !exists {
+			continue
+		}
+
+		err = DecompressDecryptBytes(&EmptyWriteIgnorer{WriteCloser: writeCloser}, archiveReader, decompressor)
```
Comment (proggga) [inline internal/wal_fetch_handler.go:144]: don't get this one, function called 'WithoutDecompressor' but we try to decompress, I'm really don't get it

Comment (prolomova) [reply internal/wal_fetch_handler.go:144]: yes, I fixed it

File: docker/redis_tests/scripts/tests/redis_full_backup_test.sh (hunk 1 above)
Comment (prolomova) [reply docker/redis_tests/scripts/tests/redis_full_backup_test.sh]: done

**Review by proggga — APPROVED** [review]

**Pull request merged**